Mathlib/RingTheory/Ideal/Basic.lean

```diff
@@ -40,20 +38,18 @@ section Semiring
 
 namespace Ideal
 
-variable [Semiring α] (I : Ideal α) {a b : α}
+variable {α : ι → Type*} [Π i, Semiring (α i)] (I : Π i, Ideal (α i))
 
 section Pi
 
-variable (ι : Type v)
-
 /-- `I^n` as an ideal of `R^n`. -/
```

Comment:
This should presumably now use a product symbol, and remark that "`I^n` as an ideal of `R^n`" is recovered as a special case?

Reply:
```suggestion
/-- `Πᵢ Iᵢ` as an ideal of `Πᵢ Rᵢ`. -/
```